### PR TITLE
[tools]add transform graph(fold_moments,fold_batch_norms_algebraic)

### PR DESCRIPTION
**Pull request by czy2014hust**
1. fold moments and batch norms when using instance_norm
2. add test case

### CODE REVIEW

**Review by petewarden — DISMISSED** [review]

**Review by petewarden — DISMISSED** [review]

**Comment by rthadur** [comment]
@czy2014hust can you please resolve conflicts ?

**Review by petewarden — APPROVED** [review]

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: tensorflow/tools/graph_transforms/fold_moments.cc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        // Find all the nodes we expect in the subgraph.
+        const NodeDef& variance_mean_node = match.node;
+        const NodeDef& mul_node = match.inputs[0].node;
+        const NodeDef& sub_node = match.inputs[0].inputs[0].node;
+        const NodeDef& mean_node = match.inputs[0].inputs[0].inputs[1].node;
+        const NodeDef& mean_node_input_node =
+            match.inputs[0].inputs[0].inputs[1].inputs[0].node;
+        const NodeDef& mean_reduction_indices_node =
+            match.inputs[0].inputs[0].inputs[1].inputs[1].node;
+        CHECK_EQ(sub_node.input(0), mean_node.input(0))
```
Comment (rthadur) [inline tensorflow/tools/graph_transforms/fold_moments.cc]: please use DCHECK instead here

File: tensorflow/tools/graph_transforms/fold_batch_norms_algebraic.cc [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        const NodeDef& gamma_node = match.inputs[0].inputs[1].inputs[1].node;
+        const NodeDef& beta_node = match.inputs[1].inputs[0].node;
+        const NodeDef& mul2_node = match.inputs[1].inputs[1].node;
+        const NodeDef& mul_node_alias =
+            match.inputs[1].inputs[1].inputs[1].node;
+
+        const NodeDef& mean_node = match.inputs[1].inputs[1].inputs[0].node;
+        const NodeDef& variance_node =
+            match.inputs[0].inputs[1].inputs[0].inputs[0].inputs[0].node;
+        CHECK_EQ(mul_node.name(), mul_node_alias.name())
```
Comment (rthadur) [inline tensorflow/tools/graph_transforms/fold_batch_norms_algebraic.cc]: please use DCHECK instead here

**Review by rthadur — CHANGES_REQUESTED** [review]
